hack/docker.sh

```diff
@@ -0,0 +1,26 @@
+#! /bin/bash
+
+git_commit=$1
+git_tag=$2
+docker_tag=edge
+
+commit_tag=$(git tag --contains ${git_commit})
+
+if [[ ${commit_tag} == ${git_tag} ]]; then
+    # we're on the tag, use the docker image for the tag
+    docker_tag=${git_tag//v/}
+    echo ${docker_tag}
+    exit 0
+
+else
+    # we're on master or a branch, try to download the latest edge and see if sha matches
+    docker pull nginx/nginx-ingress:${docker_tag} >/dev/null 2>&1
```

Comment:
do we rebuild the edge images on every commit in the master or we optimize (for example, when no code changes were in the commit?)? if latter is the case, then we might not have an image that matches the commit

Reply:
We rebuild the images for every change except for docs and examples. In that case, you're right, it won't be possible to download the binary. Unfortunately, I don't see a workaround for this unless we want to always use the latest available binary, but I think that could cause more issues.

Similarly, if somebody is on an old commit, there's no way to get the corresponding `edge` image that was built in the past (unless we start tagging the image with the sha).

Reply:
if this can cause problems, perhaps we shall only support downloading for release tags then?

Reply:
if it's not the latest commit or a tag it will print this message:
> unable to build with TARGET=download, this function is only available when building from a git tag or from the latest commit matching the edge image

do you it's not enough?

Reply:
> do you it's not enough?

yeah, I think it is enough